apis/kueue/v1beta1/localqueue_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LocalQueueSpec defines the desired state of LocalQueue
+type LocalQueueSpec struct {
+	// clusterQueue is a reference to a clusterQueue that backs this localQueue.
+	ClusterQueue ClusterQueueReference `json:"clusterQueue,omitempty"`
+}
+
+// ClusterQueueReference is the name of the ClusterQueue.
+type ClusterQueueReference string
+
+// LocalQueueStatus defines the observed state of LocalQueue
+type LocalQueueStatus struct {
+	// PendingWorkloads is the number of Workloads in the LocalQueue not yet admitted to a ClusterQueue
+	// +optional
+	PendingWorkloads int32 `json:"pendingWorkloads"`
+
+	// AdmittedWorkloads is the number of workloads in this LocalQueue
+	// admitted to a ClusterQueue and that haven't finished yet.
+	// +optional
+	AdmittedWorkloads int32 `json:"admittedWorkloads"`
```

Comment:
Ideally, there should be an information about about resource consumption of these workloads. At this moment only part of the information from ClusterQueue is propagated here.

Reply:
Yes, but I'll leave this for future improvements #542